Rrefactor `defineWpConfigConsts` to work on virtual file system

sejas opened the pull request:
<!-- Thanks for contributing to WordPress Playground! -->

## What?

<!-- In a few words, what is the PR actually doing? Include screenshots or screencasts if applicable -->

Remove `defineVirtualWpConfigConsts` in favor of `defineWpConfigConsts`, adding a `virtualize` option.

## Why?

<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

Code optimization discussed here https://github.com/WordPress/wordpress-playground/pull/343#discussion_r1195701721

## Testing Instructions

1. Observe the blueprint tests pass
2. Run `nx preview wp-now start --path=/path/to/wordpress-plugin-or-theme`
3. Observe there is no `playground-consts.json` file added to the root of wordpress.
4. Observe the wp-config.php was not modified.

adamziel commented on packages/playground/blueprints/src/lib/steps/define-wp-config-consts.ts:
```diff
@@ -8,42 +8,46 @@ export interface DefineWpConfigConstsStep {
+	let basePath = documentRoot;
+	if (virtualize) {
+		playground.mkdir(VFS_CONFIG_FILE_BASENAME);
+		basePath = VFS_CONFIG_FILE_BASENAME;
```
Yay! By the way, any reason not to set the php.ini option here? Otherwise the executor needs to do some reasoning based on the steps.

adamziel replied:
@sejas ping ^ :-)

sejas replied:
@adamziel ,
Ha!, thanks for the suggestion. Really smart!
https://github.com/WordPress/wordpress-playground/pull/415/commits/41b332fc3a2ae103e04b1e8811190c93cee603c2

adamziel commented:
Thank you @sejas! Something is wrong with the formatting, though. I added a husky pre-commit hook to run `nx format` but I guess it doesn't auto-install 🤔  any ideas how to fix that?

sejas commented:
> Thank you @sejas! Something is wrong with the formatting, though. I added a husky pre-commit hook to run `nx format` but I guess it doesn't auto-install 🤔 any ideas how to fix that?

Fixed! https://github.com/WordPress/wordpress-playground/pull/415/commits/5944df3b254d429b724dcdab406f19fda674a880

sejas commented:
@adamziel, can I merge this branch?

adamziel commented on packages/playground/blueprints/src/lib/steps/define-wp-config-consts.ts:
```diff
@@ -1,5 +1,5 @@
 import { StepHandler } from '.';
-import { updateFile } from './common';
+import { VFS_CONFIG_FILE_BASENAME, updateFile } from './common';
```
Could this constant be defined in this file?

adamziel replied:
Also, how about `VFS_TMP_DIRECTORY` as a name, since that seems to be the intent?

sejas replied:
done! https://github.com/WordPress/wordpress-playground/pull/415/commits/2e9b0dd352717685917d01c00ceb4b34c519f0d1

sejas commented:
@adamziel, I would appreciate a final review of this PR. Thank you!

adamziel commented on packages/playground/blueprints/src/lib/steps/define-wp-config-consts.ts:
```diff
@@ -8,42 +8,46 @@ export interface DefineWpConfigConstsStep {
 	step: 'defineWpConfigConsts';
 	/** The constants to define */
 	consts: Record<string, unknown>;
+	/** Whether to virtualize the wp-config.php and playground-consts.json files, by default false */
```
Let’s explain what virtualization means here and how it won’t affect the local project files

sejas replied:
done! https://github.com/WordPress/wordpress-playground/pull/415/commits/ad89004cdcce4a5395cb209567c19193489977c1

adamziel reviewed (commented):
Almost there, I only left a few nit picks. Thank you for working on this!

adamziel commented:
Thank you for consolidating that logic @sejas!
